src/crystal/system/unix/user.cr

```diff
@@ -0,0 +1,45 @@
+require "c/pwd"
+
+module Crystal::System::User
+  private def from_struct(pwd)
+    user = String.new(pwd.pw_gecos).split(",").first
+    new(String.new(pwd.pw_name), pwd.pw_uid.to_s, pwd.pw_gid.to_s, user, String.new(pwd.pw_dir), String.new(pwd.pw_shell))
+  end
+
+  def from_username?(username : String)
+    username.check_no_null_byte
+
+    pwd = uninitialized LibC::Passwd
+    pwd_pointer = pointerof(pwd)
+    buf = Bytes.new(1024)
+
+    ret = LibC.getpwnam_r(username, pwd_pointer, buf, buf.size, pointerof(pwd_pointer))
+    while ret == LibC::ERANGE
```

Comment:
It's allocating 2× more memory in a loop. I assume the buffer should eventually be large enough, but I'm not fond of such an assumption, and would add an upper limit (e.g. 16KB) and raise Errno if it still fails.

Reply:
Good call, I'll add a test to the loop. The `raise` that follows the loop should suffice, since `ret` should be `ERANGE`.